crates/tracing-instrumentation/src/lib.rs

```diff
@@ -129,21 +148,21 @@ where
 
     let provider = tracer_provider_builder.build();
 
-    let tracer = SpanModifyingTracer::new(
-        provider
-            .tracer_builder("opentelemetry-otlp")
-            .with_version(env!("CARGO_PKG_VERSION"))
-            .build(),
-    );
-    let _ = opentelemetry::global::set_tracer_provider(provider);
+    let tracer = provider
+        .tracer_builder("opentelemetry-otlp")
+        .with_version(env!("CARGO_PKG_VERSION"))
+        .build();
+
+    opentelemetry::global::set_tracer_provider(provider);
 
     Ok(Some(
         tracing_opentelemetry::layer()
             .with_location(false)
             .with_threads(false)
             .with_tracked_inactivity(false)
             .with_tracer(tracer)
-            .with_filter(EnvFilter::try_new(&common_opts.tracing.tracing_filter)?),
+            .with_filter(EnvFilter::try_new(&opts.tracing_filter)?)
+            .boxed(),
```

Comment:
I had to simplify the return typed to Box<dyn> type to make it easier to change the layer setup without the need to change the function signature every time I add a new .with! 
This was helpful during development. But i think we can revert to a concrete type back since this won't be needed anymore